LoRaEngine/LoraKeysManagerFacade/CreateEdgeDevice.cs

```diff
@@ -146,6 +148,40 @@ public CreateEdgeDevice(RegistryManager registryManager)
             return PrepareResponse(HttpStatusCode.OK);
         }
 
+        private static string ReplaceJsonWithCorrectValues(string region, string resetPin, string json, string spiSpeed, string spiDev)
+        {
+            json = json.Replace("[$region]", region);
+            json = json.Replace("[$reset_pin]", resetPin);
+
+            if (string.Equals(spiSpeed, "8", StringComparison.CurrentCultureIgnoreCase) ||
+                string.IsNullOrEmpty(spiSpeed))
+            {
+                // default case
+                json = json.Replace("[$spi_speed]", string.Empty);
+            }
+            else
+            {
+                json = json.Replace(
+                    "[$spi_speed]",
+                    ",'SPI_SPEED':{'value':'2'}");
+            }
+
+            if (string.Equals(spiDev, "2", StringComparison.CurrentCultureIgnoreCase) ||
+                string.IsNullOrEmpty(spiDev))
+            {
+                // default case
+                json = json.Replace("[$spi_dev]", string.Empty);
+            }
+            else
+            {
+                json = json.Replace(
+                    "[$spi_dev]",
+                    ",'SPI_DEV':{'value':'1'}");

```

Comment:
same with spiDev. Right now, we never add the passed in value to the json string. If it is 2 the result will be empty, in all other cases it will be one, no matter what value was passed in. If that's really the desired behavior, it needs comments.

Reply:
Looking at the ARM template further down, it looks like there are only 2 values allowed. Need to clarify.